code/java/src/main/java/io/swagger/client/model/ZakelijkGerechtigdeHal.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Kadastraal onroerende zaken
+ * D.m.v. deze toepassing worden meerdere, korte bevragingen op de Basis Registratie Kadaster beschikbaar gesteld. Deze toepassing betreft het verstrekken van Kadastrale Onroerende Zaak informatie.
+ *
+ * OpenAPI spec version: 1.0.0
+ * 
+ *
+ * NOTE: This class is auto generated by the swagger code generator program.
+ * https://github.com/swagger-api/swagger-codegen.git
+ * Do not edit the class manually.
+ */
+
+package io.swagger.client.model;
+
+import java.util.Objects;
+import java.util.Arrays;
+import com.google.gson.TypeAdapter;
+import com.google.gson.annotations.JsonAdapter;
+import com.google.gson.annotations.SerializedName;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import io.swagger.client.model.ErfpachtCanon;
+import io.swagger.client.model.PersoonBeperkt;
+import io.swagger.client.model.Tenaamstelling;
+import io.swagger.client.model.TypeGerechtigdeEnum;
+import io.swagger.client.model.ZakelijkGerechtigde;
+import io.swagger.client.model.ZakelijkGerechtigdeLinks;
+import io.swagger.v3.oas.annotations.media.Schema;
+import java.io.IOException;
+import org.threeten.bp.LocalDate;
```

Comment:
[Backport](https://www.threeten.org/threetenbp/) van Java 8 date-time API? Is genereren voor Java 8 niet mogelijk? Want Java 7 is _end of life_ sinds juli 2015.

Reply:
Het valt mij überhaupt op dat de gegenereerde Java code er iets anders uiziet dan de implementatie van het Kadaster. Jackson ipv Gson voor XML 🤷‍♂️ 

Mocht het interessant zijn, wij gebruiken de [openapi-generator-maven-plugin](https://openapi-generator.tech/docs/plugins/) voor het genereren van de Java code. Voor Java 8 `<java8>true</java8>` en met Java 8 date-time API `<dateLibrary>java8</dateLibrary>`.  
_Overigens bevat versie 4.3.1 [een bug met allOf inheritance](https://github.com/OpenAPITools/openapi-generator/issues/6815), gebruik dus versie 4.3.0._

Reply:
@rhengeveld gebruiken jullie openapi-generator generate ook voor een client?

`openapi-generator generate -g java -i ../specificatie/BRK-Bevragen/genereervariant/openapi.yaml -o /java --additional-properties=dateLibrary=java8,java8=true`

Reply:
Ah, nee, we hebben momenteel geen _client_ implementatie. Ik zou echter verwachten dat de gegenereerde classes bruikbaar zijn ongeacht of het een _server_ of _client_ is 🤔

Overigens activeren we nog een aantal andere settings op onze _server_, o.a. `<library>spring-boot</library>` en `<useBeanValidation>true</useBeanValidation>`. Dat is echter implementatie afhankelijk; wij gebruiken het Spring Boot framework en willen diens automatische validatie gebruiken.

Reply:
@rhengeveld ik heb de java client nu gegenereerd met openapi-generator. Zou jij hier naar willen kijken of dit geleid heeft tot goede client code?

Reply:
Wow, dat is behoorlijk veel code. Ik zag iets vreemds met de `BadRequestFoutberichtAllOf` (zie andere opmerking) waarvan ik niet zeker weet of dat nog steeds van toepassing is. Tevens vraag ik mij af of de generator versie 1.0.0 is of misschien toch 4.3.0. 

Buiten dat:
* ✔️ `java.util.time` imports voor Java 8 date-time API
* ✔️ support voor Java 8 en hoger  
_(Java 7 is vervallen, maar daar daar kunnen we hopelijk wel mee leven 😉)_  

Zo snel ik het kan overzien ziet het er goed uit 👍

Reply:
@rhengeveld 
> Tevens vraag ik mij af of de generator versie 1.0.0 is of misschien toch 4.3.0.

Ik heb inderdaad 4.3.0 gebruikt. Had uit je eerdere opmerking geconcludeerd dat ik die moet gebruiken en niet 4.3.1.

Had ik 1.0.0 moeten installeren?

Reply:
Versie `4.3.0` is de goede versie 👍